test/functional/core/job_spec.lua

```diff
@@ -215,27 +244,40 @@ describe('jobs', function()
   end)
 
   it('avoids sending final newline', function()
-    if helpers.pending_win32(pending) then return end  -- TODO: Need `cat`.
-    nvim('command', "let j = jobstart(['cat', '-'], g:job_opts)")
+    nvim('command', "let j = jobstart(has('win32') ? ['find', '/v', ''] : ['cat', '-'], g:job_opts)")
     nvim('command', 'call jobsend(j, ["some data", "without\nfinal nl"])')
-    eq({'notification', 'stdout', {0, {'some data', 'without\nfinal nl'}}},
-      next_msg())
+    if iswin() then
+      expect_msg_seq(
+        { {'notification', 'stdout', {0, {'some data', 'without', ''}}},
+        },
+        -- Alternative sequence:
+        { {'notification', 'stdout', {0, {'some data', ''}}},
+          {'notification', 'stdout', {0, {'without', ''}}},
+        }
+      )
+    else
+      eq({'notification', 'stdout',
+        {0, {'some data', 'without\nfinal nl'}}}, next_msg())
+    end
     nvim('command', "call jobstop(j)")
+    if iswin() then
+      eq({'notification', 'stdout', {0, {'final nl', ''}}}, next_msg())
```

Comment:
This output has the newline, doesn't it? That defeats the purpose of the test. Should just be pending.

Reply:
It's leftover from stdout when find.exe exits for some reason. Don't recall this happening on a regular cmd.exe window. Strange that this doesn't happen on `jobclose(g:job_id, "stdin")`.